jobsdb/jobsdb.go

```diff
@@ -1240,37 +1233,26 @@ func mapDSToLevel(ds dataSetT) (levelInt int, levelVals []int, err error) {
 	return len(levelVals), levelVals, nil
 }
 
-func (jd *HandleT) createTableNames(dsIdx string) (string, string) {
+func (jd *HandleT) newDataSetStruct(dsIdx string) dataSetT {
 	jobTable := fmt.Sprintf("%s_jobs_%s", jd.tablePrefix, dsIdx)
 	jobStatusTable := fmt.Sprintf("%s_job_status_%s", jd.tablePrefix, dsIdx)
-	return jobTable, jobStatusTable
+	return dataSetT{
+		JobTable:       jobTable,
+		JobStatusTable: jobStatusTable,
+		Index:          dsIdx,
+	}
 }
 
-func (jd *HandleT) addNewDS(newDSType string, insertBeforeDS dataSetT) dataSetT {
-	jd.logger.Infof("Creating new DS of type %s before ds %s for %s jobsdb", newDSType, insertBeforeDS.Index, jd.tablePrefix)
-	var newDSIdx string
-	appendLast := newDSType == appendToDsList
+func (jd *HandleT) addDS(ds dataSetT, isNew bool) {
+	jd.logger.Infof("Creating new DS %+v", ds)
 
 	queryStat := stats.NewTaggedStat("add_new_ds", stats.TimerType, stats.Tags{"customVal": jd.tablePrefix})
 
 	queryStat.Start()
 	defer queryStat.End()
 
-	switch newDSType {
-	case appendToDsList:
-		newDSIdx = jd.computeNewIdxForAppend()
-	case insertForMigration:
-		newDSIdx = jd.computeNewIdxForIntraNodeMigration(insertBeforeDS)
-	case insertForImport:
-		newDSIdx = jd.computeNewIdxForInterNodeMigration(insertBeforeDS)
-	default:
-		panic("Unknown usage for newDSType : " + newDSType)
```

Comment:
Just making sure:
We no longer need this, since we directly pass the index when calling `addDs`. Correct?

Reply:
correct, index calculation is performed earlier